### DISCUSSION TITLE
Does BrowserGym supports image_som?

### DISCUSSION THREAD

**Discussion by junleiz** [ServiceNow/BrowserGym#116] [General]
I do not find image_som in the current obs:
        obs = {
            "chat_messages": copy.deepcopy(self.chat.messages),
            "goal": goal_msg,  # TODO: redundant with chat messages, to be removed?
            "goal_image_urls": goal_image_urls,  # TODO: redundant with chat messages, to be removed?
            "open_pages_urls": [page.url for page in self.context.pages],
            "active_page_index": np.asarray([self.context.pages.index(self.page)]),
            "url": self.page.url,
            "screenshot": extract_screenshot(self.page),
            "dom_object": dom,
            "axtree_object": axtree,
            "extra_element_properties": extra_properties,
            "focused_element_bid": focused_element_bid,
            "last_action": self.last_action,
            "last_action_error": self.last_action_error,
            "elapsed_time": np.asarray([time.time() - self.start_time]),
        }
        Does BrowerGym supports image som ?

**Comment by TLSDC** [comment]
Hi @leoozy, BGym does have a function to generate the SOM representation.
https://github.com/ServiceNow/BrowserGym/blob/87c45ad46a9f7b992f286383bf04c54c53c9882e/browsergym/core/src/browsergym/utils/obs.py#L425-L431
It is not computed by default to not slow down text only users.
Hope this helps :)

**Comment by gasse** [comment]
See here for an example
https://github.com/ServiceNow/AgentLab/blob/f385533ce85b89ad9456f676ebd019221d968f85/src/agentlab/agents/dynamic_prompting.py#L819
